src/bin/vhost_user_fs.rs

```diff
@@ -169,7 +183,23 @@ impl<F: FileSystem + Send + Sync + 'static> VhostUserBackend for VhostUserFsBack
             REQ_QUEUE_EVENT => {
                 debug!("REQ_QUEUE_EVENT");
                 let mut vring = vrings[1].write().unwrap();
-                self.process_queue(&mut vring)?;
+                if self.event_idx {
+                    // vm-virtio's Queue implementation only checks avail_index
```

Comment:
Should change vm-virtio?

Reply:
Actually, vm-virtio's Queue behavior is quite reasonable when not using EVENT_IDX. I think it's better to leave up to the callers to decide whether they want to check avail_index again by generating a new Queue iterator, or just do it once.